src/providers/s3.js

```diff
@@ -65,11 +74,36 @@ class S3Provider extends BaseProvider {
         if (err) {
           return reject(err);
         }
-        return resolve(result);
+        return resolve({
+          invalidate: this.invalidate,
+          result
+        });
       });
     });
   }
 
+  invalidate(invalidationPaths = ['/*'], distribution = process.env.CLOUDFRONT_DISTRIBUTION_ID) {
+    return new Promise((resolve, reject) => {
+      assert(distribution, 'distribution has to be a non empty string');
+      assert(
+        (invalidationPaths && typeof invalidationPaths === 'string')
+        || Array.isArray(invalidationPaths),
+        'invalidationPaths must be a string or array of strings'
+      );
+      const targetPaths = Array.isArray(invalidationPaths) ? invalidationPaths : [invalidationPaths];
```

Comment:
you already did a check for Array.isArray(invalidationPaths) reuse it

Reply:
I am not sure I understand how I can reuse it. The previous check is inside the assertion to validate if it's either a string or an array. Then, I convert a string to array to make it easier to process later

Reply:
hm something like this then:
```js
const isArray = Array.isArray(invalidationPaths);
assert( typeof invalidationPaths === 'string' || isArray)
const targetPaths = isArray ? invlidationPaths : [invalidationPaths]
```